### ISSUE TITLE
rename device

### ISSUE THREAD

**Issue by lucaspinelli85** [mongoose-os-apps/shelly-homekit#627]
Is possible to rename the device to make more discoverable in the wireless client id in my airport?
![IMG_5314](https://user-images.githubusercontent.com/74553551/114322455-a423cb00-9b20-11eb-995e-264ebc8db60a.PNG)
<img width="656" alt="image" src="https://user-images.githubusercontent.com/74553551/114322510-eea54780-9b20-11eb-8f84-20eeafcddff3.png">

**Comment by andyblac** [comment]
i take it they are all brand new devices, that have not been setup yet ?

**Comment by andyblac** [comment]
if theyhave been setup and connected to your wifi,you can cnage the name in General settings
https://github.com/mongoose-os-apps/shelly-homekit/wiki/General-Settings

**Comment by lucaspinelli85** [comment]
i already changed the name, but the one displayed is the id
<img width="489" alt="image" src="https://user-images.githubusercontent.com/74553551/114323213-8a848280-9b24-11eb-97b4-c676f727672f.png">
<img width="464" alt="image" src="https://user-images.githubusercontent.com/74553551/114323238-983a0800-9b24-11eb-9bcd-07e9f1eb2664.png">
<img width="263" alt="image" src="https://user-images.githubusercontent.com/74553551/114323253-a5ef8d80-9b24-11eb-8247-12d9513e745e.png">

**Comment by lucaspinelli85** [comment]
So the real question is: is it possible to rename the device id?

**Comment by andyblac** [comment]
> So the real question is: is it possible to rename the device id?

yes, device id is hardcoded to the device, the option above changes the hostname of the device.

**Comment by andyblac** [comment]
@rojer is there another dns option that needs tweaking to use hostname ?

**Comment by timoschilling** [comment]
> yes, device id is hardcoded to the device

That's not true, it is changeable via the config key `device.id`, but maybe that can have side effects. @andyblac what you mean is the chip id and the chip id is part of the device id.

@lucaspinelli85 there is somewhere a bug in the dns setup because of that routes are showing the device id and not the hostname.

I can't find the issue, but if I remind it right there is one place in the dns setup where we use the device id and not the hostname.

**Comment by andyblac** [comment]
> > yes, device id is hardcoded to the device
> 
> That's not true, it is changeable via the config key `device.id`, but maybe that can have side effects. @andyblac what you mean is the chip id and the chip id is part of the device id.
> 
> @lucaspinelli85 there is somewhere a bug in the dns setup because of that routes are showing the device id and not the hostname.
> 
> I can't find the issue, but if I remind it right there is one place in the dns setup where we use the device id and not the hostname.

i meant not user changable ;)

**Comment by lucaspinelli85** [comment]
For example, the iPhone, the MacBook, and the HomePod are showed by his hostname! Only the shelly are showed by device id and not hostname in wireless client list and make difficult identify them

**Comment by timoschilling** [comment]
Changing the device id is the wrong way, mongoose need to fix the dhcp request. See last line of my previous comment. 

@rojer can you fix that or give me a hint?

**Comment by lucaspinelli85** [comment]
so to see the host name instead of the device id in the wireless client list I have to wait for a fix on a next update? correct?

**Comment by andyblac** [comment]
> so to see the host name instead of the device id in the wireless client list I have to wait for a fix on a next update? correct?

correct, wait for a fix in mongoose.

**Comment by lucaspinelli85** [comment]
Ok i'll wait it!

**Comment by zhoucky** [comment]
Hi, my Linksys router is showing the hostnames in the device list. In the summerhouse I have D-link, it’s also showing the hostnames.

**Comment by lucaspinelli85** [comment]
I keep seeing the device-id instead of the device-name

**Comment by rojer** [comment]
we use names that are guaranteed to be unique until name conflic resolution is implemented.
see https://github.com/mongoose-os-apps/shelly-homekit/commit/acb57e02caef82438a2c82da5c67265a8c76fda0

**Comment by lucaspinelli85** [comment]
> we use names that are guaranteed to be unique until name conflic resolution is implemented.
> 
> see https://github.com/mongoose-os-apps/shelly-homekit/commit/acb57e02caef82438a2c82da5c67265a8c76fda0

ok, it's not an urgent serious problem! Better to guarantee uniqueness!
